src/services/attendee/attendee-router.ts

```diff
@@ -23,22 +23,31 @@ attendeeRouter.post(
     async (req, res) => {
         const payload = res.locals.payload;
         const userId = payload.userId;
+
         const { eventId } = EventIdValidator.parse(req.params);
 
-        const attendee = await Database.ATTENDEE.findOne({ userId });
+        const { data: attendee } = await SupabaseDB.ATTENDEES.select(
+            "favoriteEvents"
+        )
+            .eq("userId", userId)
+            .maybeSingle()
+            .throwOnError();
 
         if (!attendee) {
             return res
                 .status(StatusCodes.NOT_FOUND)
                 .json({ error: "UserNotFound" });
         }
 
-        await Database.ATTENDEE.updateOne(
-            { userId: userId },
-            { $addToSet: { favorites: eventId } }
-        );
+        const newFavorites = (attendee.favoriteEvents || []).includes(eventId)
+            ? attendee.favoriteEvents || []
+            : [...(attendee.favoriteEvents || []), eventId];
```

Comment:
this can definitely be cleaner - imo preset the array to something like this:

```
var favs = attendee.favoriteEvents || [];
if (!favs.includes(eventId)) {
// push
} 
```